src/pki/pkcs.ts

```diff
@@ -0,0 +1,276 @@
+/** based on document Gemeinsame Grundsätze Technik, Anlage 16
+ * (see /docs/documents.md for more info)
+ */
+
+import EncapsulatedContentInfo from "pkijs/src/EncapsulatedContentInfo";
+import SignedData from "pkijs/src/SignedData";
+import SignerInfo from "pkijs/src/SignerInfo";
+import AlgorithmIdentifier from "pkijs/src/AlgorithmIdentifier";
+import IssuerAndSerialNumber from "pkijs/src/IssuerAndSerialNumber";
+import RSASSAPSSParams from "pkijs/src/RSASSAPSSParams";
+import EnvelopedData from "pkijs/src/EnvelopedData";
+import ContentInfo from "pkijs/src/ContentInfo";
+import CertificationRequest from "pkijs/src/CertificationRequest";
+import AttributeTypeAndValue from "pkijs/src/AttributeTypeAndValue";
+import Certificate from "pkijs/src/Certificate";
+import { OctetString, PrintableString, fromBER } from "asn1js";
+import { initCrypto } from "./crypto";
+import { bufferToCertificate, importPKCS8, exportPKCS8, bufferToHex } from "./utils";
+import { File, ResultOrErrors } from "../types";
+import { isIK, isTruncatedIfTooLong, isVarchar, validationByType } from "../validation/utils";
+import { transliterateCertificateName, transliterateRecursively } from "../transcoding";
+
+// - PKCS#7 methods
+
+export const signAndEncryptMessage = async (
+    message: ArrayBuffer,
+    signerCertificate: ArrayBuffer,
```

Comment:
couldn't this already be a `Certificate` object? The process gets its certificate from the `InstitutionListsIndex` and that class already needs the `Certificate`s as objects because it needs to check if they are still valid by date.

Probably this is related to that the `InstitutionListsIndex` uses a different dependency for certificate parsing.

Reply:
I consider the pkijs `Certificate` object an implementation detail tied to the pkijs library. I want `signAndEncryptMessage` to be part of PAID's public API and for these exposed methods I prefer to hand over certificates in standardized DER format, despite being less efficient when the certificate is converted back and forth.